sounddevice.py

```diff
@@ -2262,13 +2262,67 @@ def reset(self):
         vars(self).clear()
         self.__init__()
 
+
 if not hasattr(_ffi, 'I_AM_FAKE'):
     # This object shadows the 'default' class, except when building the docs.
     default = default()
 
 
 class PortAudioError(Exception):
-    """This exception will be raised on PortAudio errors."""
+    """This exception will be raised on PortAudio errors.
+
+    Parameters
+    ----------
+    msg : str, optional
+        User error message.
+    code : int, optional
+        PaErrorCode value or None if there is no PaErrorCode associated
+        with the error.
+
+    """
+
+    code = None
+    """The PaErrorCode integer value or None if there is no associated error
+    code.
+
+    """
+    host_api = None
+    """The index of the host API which returned the error code or None if the
+    error is not a host API error.
+
+    """
+    hosterror_info = None
+    """Either a tuple containing the host error code and a textual description
+    of the host error or None if the error is not a host API error. The host
+    error description will be a zero length string if one is not available.
+
+    """
+
+    def __init__(self, msg=None, code=None):
+        self.code = code
+        self.hosterror_info = self.host_api = None
+
+        if code:
+            errormsg = _ffi.string(_lib.Pa_GetErrorText(code)).decode()
+            if code == _lib.paUnanticipatedHostError:
+                info = _lib.Pa_GetLastHostErrorInfo()
+                self.host_api = _lib.Pa_HostApiTypeIdToHostApiIndex(info.hostApiType)
+                hosterror_text = _ffi.string(info.errorText).decode()
+                hosterror_code = info.errorCode
+                self.hosterror_info = hosterror_code, hosterror_text
+                hostname = query_hostapis(self.host_api)['name']
+                errormsg += " [PaErrorCode {0}, {1}ErrorCode {2}]"\
```

Comment:
Thanks for changing this, but there is no such thing as a `XyzErrorCode`, which your spelling would suggest, or is there?
Can't you just use `XYZ error 99` or `XYZ error code 99`, or really just anything without CamelCase?

Just to be clear: `PaErrorCode` is fine, because this actually exists in the PortAudio API.